### PR TITLE
Snow Leopards - Elsje S

### PR DESCRIPTION
**Pull request by elsjeslothower**

### CODE REVIEW

**Comment by elsjeslothower** [comment]
TLDR: I changed a test that didn't indicate it needed any modification.

The git commit that says "very confused...reminder to come back to this" was when I had to change a test in order for it to pass. As far as I could tell, the feature that was being tested didn't need to run 1000 times; I triple-checked my code's logic and went line by line through the test code and realized it was fulfilling what it needed to with one iteration. Let me know if I missed anything, or if there is a specific reason behind putting the test in a loop. (a little more added context: the expected value became undefined at some point in the looping)

File: src/adagrams.js [hunk 1]
```diff
@@ -1,15 +1,113 @@
+const letterDistribution = ['A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A',
```
Comment (marciaga) [inline src/adagrams.js:1]: I don't think there was any prohibition on doing it this way, but generating the distribution programmatically would have been better

File: src/adagrams.js [hunk 2]
```diff
@@ -1,15 +1,113 @@
...
+  'J': 8,
+  'X': 8,
+  'Q': 10,
+  'Z': 10
+}
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+  const letterPool = letterDistribution;
```
Comment (marciaga) [inline src/adagrams.js:39]: Since this function mutates `letterPool`, it also mutates `letterDistribution`, so there wasn't a need to assign it in this case.

File: src/adagrams.js [hunk 3]
```diff
@@ -1,15 +1,113 @@
...
+  const letterBank = [];
+  
+  for (let i=0; i<10; i++) {
+    let randomIndex = Math.floor(Math.random() * letterPool.length);
+    let randomLetter = letterPool[randomIndex];
+    letterBank.push(randomLetter);
```
Comment (marciaga) [inline src/adagrams.js:45]: This utilizes the distribution correctly, nice!

File: src/adagrams.js [hunk 4]
```diff
@@ -1,15 +1,113 @@
...
+    letterPool.splice(randomIndex, 1);
+  }
+  return letterBank;
+  };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const letterBank = lettersInHand.slice();
```
Comment (marciaga) [inline src/adagrams.js:52]: Nice job with this one!

File: src/adagrams.js [hunk 5]
```diff
@@ -1,15 +1,113 @@
...
+  if (challenger) {
+    const champion = tiebreaker(challenger, winningPair);
+    return champion;
+  }
+
+  return winningPair;
 };
+
+const tiebreaker = (challenger, winningPair) => {
+  if (challenger['word'].length === winningPair['word'].length) {
```
Comment (marciaga) [inline src/adagrams.js:102]: generally prefer using dot notation for accessing object properties

File: src/adagrams.js [hunk 6]
```diff
@@ -1,15 +1,113 @@
...
+      highScore = score;
+    } else if (score === highScore) {
+      challenger = {'word': words[challengerCounter], 'score': score};
+    }
+    challengerCounter += 1;
+  }
+
+  const scoreIndex = wordScores.indexOf(highScore);
+  const winningWord = words[scoreIndex];
+  const winningPair = {'word': winningWord, 'score': highScore};
```
Comment (marciaga) [inline src/adagrams.js:91]: JS Object keys should be written as symbols, i.e.
```javascript
{ word: winningWord, score: highScore }
```

File: test/adagrams.test.js [hunk 7]
```diff
@@ -133,22 +134,22 @@ describe("Adagrams", () => {
...
 
-    it("accurately finds best scoring word even if not sorted", () => {
-      const words = ["XXX", "XXXX", "X", "XX"];
-      const correct = { word: "XXXX", score: scoreWord("XXXX") };
+    it('accurately finds best scoring word even if not sorted', () => {
+      const words = ['XXX', 'XXXX', 'X', 'XX'];
+      const correct = { word: 'XXXX', score: scoreWord('XXXX') };
 
-      throw "Complete test by adding an assertion";
+      expect(highestScoreFrom(words)).toEqual(correct);
```
Comment (marciaga) [inline test/adagrams.test.js:149]: Well done!

File: test/adagrams.test.js [hunk 8]
```diff
@@ -51,79 +51,80 @@ describe("Adagrams", () => {
...
       expectScores({
         a: 1,
         dog: 5,
         wHiMsY: 17,
       });
     });
 
-    it("returns a score of 0 if given an empty input", () => {
-      throw "Complete test";
+    it('returns a score of 0 if given an empty input', () => {
```
Comment (marciaga) [inline test/adagrams.test.js:121]: Nice!

**Review by marciaga — COMMENTED** [review]
Great job overall, a few small comments here and there!
